File: FinanceSystemAPI/Models/Operation.cs
﻿namespace FinanceSystemAPI.Models
{
    public class Operation
    {
        public int Id { get; set; }
        public string Username { get; set; }
        public DateTime Date { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
        public double Value { get; set; }
        public string CurrencyCode { get; set; }
        public double CurrentValueInPLN { get; set; }
    }
}

File: UI/FinanceSystemUI/src/app/components/models/operation.ts
export class Operation {
    id: number;
    username: string;
    date: string;
    category: string;
    description: string;
    value: any;
    currencyCode: string;
    currentValueInPLN: any;

    constructor(id: number, username: string, date: string, category: string, description: string, value: string, currencyCode: string, currentValueInPLN: string) {
        this.id = id;
        this.username = username;
        this.date = date;   
        this.category = category;
        this.description = description;
        this.value = parseFloat(value);
        this.currencyCode = currencyCode;
        this.currentValueInPLN = parseFloat(currentValueInPLN);
    }
}
